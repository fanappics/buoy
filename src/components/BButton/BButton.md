Sample BButton example:

```js
<b-button>Buoy Button</b-button>
```
