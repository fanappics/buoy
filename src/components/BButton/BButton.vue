<template>
  <button
    :aria-label="'aria-label'"
    :disabled="disabled"
    :style="computedStyle"
    :class="computedClass"
  >
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'b-button',
  props: {
    'aria-label': {
      type: String,
      required: false
    },
    disabled: {
      type: Boolean,
      required: false,
      default: false
    },
    height: {
      type: String,
      required: false
    },
    type: {
      type: String,
      required: false,
      default: 'button'
    },
    width: {
      type: String,
      required: false
    }
  },
  data () {
    return {}
  },
  computed: {
    computedClass () {
      // if any classes are passed through, provided them to the component as an array of strings
      const classArray = this.class ? this.class.split(' ') : [];

      // can do differential class application in the future here, i.e.,
      // we may want to break out 'color' as a prop to improve readability:
      // if (this.color && !classArray.includes(this.color)) {
        // classArray.push(this.color)
      // }

      return classArray
    },
    computedStyle () {
      // allow size to be adjusted via props
      const styleObject = {}
      if (this.height) {
        styleObject.height = this.height
      }
      if (this.width) {
        styleObject.width = this.width
      }
      return styleObject
    }
  }
}
</script>
