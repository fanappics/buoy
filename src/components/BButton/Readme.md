external b-button documentation:

```js
<b-button>Buoy Button</b-button>
```
