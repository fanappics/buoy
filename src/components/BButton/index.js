import BButton from './BButton.vue';

/* istanbul ignore next */
BButton.install = function install (Vue) {
  Vue.component(BButton.name, BButton);
};

export default BButton;
