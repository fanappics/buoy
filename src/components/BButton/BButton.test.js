import { mount, shallow } from 'vue-test-utils'
import BButton from './BButton'

describe('BButton', () => {
  it('renders a button', () => {
    const wrapper = mount(BButton, {
      slots: {
        default: 'hello world'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('puts height and width props into style', () => {
    const wrapper = shallow(BButton, {
      propsData: {
        height: '50px',
        width: '100px'
      }
    })

    expect(wrapper.hasStyle('height', '50px')).toBe(true)
    expect(wrapper.hasStyle('width', '100px')).toBe(true)
  })

  it('is enabled when disabled prop is not passed', () => {
    const wrapper = shallow(BButton, {})

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('is disabled when disabled prop is passed', () => {
    const wrapper = shallow(BButton, {
      propsData: {
        disabled: true
      }
    })

    expect(wrapper.props().disabled).toBe(true)
  })

  it('has an aria-label if one is passed', () => {
    const wrapper = shallow(BButton, {
      propsData: {
        'aria-label': 'submit button'
      }
    })

    expect(wrapper.props()['aria-label']).toBe('submit button')
  })
})
