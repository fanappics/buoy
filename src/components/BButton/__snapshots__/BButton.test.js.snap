// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BButton does not render label when iconOnly true 1`] = `
<button id="button-iconOnly" title="iconOnly button" class="b-button b-primary">test

</button>
`;

exports[`BButton renders a button 1`] = `
<button id="button-1" title="" class="b-button b-primary">
    button test
</button>
`;

exports[`BButton renders a disabled button 1`] = `
<button id="button-disabled" title="" disabled="disabled" class="b-button b-primary">
    disabled
</button>
`;

exports[`BButton renders a secondary button 1`] = `
<button id="button-secondary" title="" class="b-button b-secondary">
    secondary
</button>
`;

exports[`BButton renders icon when one is passed 1`] = `
<button id="button-icon" title="" class="b-button b-primary">test icon button
</button>
`;
